pkg/declcfg/declcfg.go

```diff
@@ -0,0 +1,83 @@
+package declcfg
+
+import (
+	"encoding/json"
+
+	"github.com/operator-framework/operator-registry/pkg/property"
+)
+
+const (
+	schemaPackage = "olm.package"
+	schemaBundle  = "olm.bundle"
+)
+
+type DeclarativeConfig struct {
+	Packages []Package
+	Bundles  []Bundle
+	Others   []Meta
+}
+
+type Package struct {
+	Schema         string `json:"schema"`
+	Name           string `json:"name"`
+	DefaultChannel string `json:"defaultChannel"`
+	Icon           *Icon  `json:"icon,omitempty"`
+	Description    string `json:"description,omitempty"`
+}
+
+type Icon struct {
+	Data      []byte `json:"base64data"`
+	MediaType string `json:"mediatype"`
+}
+
+type Bundle struct {
+	Schema        string              `json:"schema"`
+	Name          string              `json:"name"`
+	Package       string              `json:"package"`
+	Image         string              `json:"image"`
+	Properties    []property.Property `json:"properties,omitempty"`
+	RelatedImages []RelatedImage      `json:"relatedImages,omitempty"`
+
+	// These fields are present so that we can continue serving
+	// the GRPC API the way packageserver expects us to in a
+	// backwards-compatible way. These are populated by a separate
+	// "objects" directory in the configs directory/tar.
+	//
+	// These fields should never be persisted in the bundle blob.
+	// Instead they will be written to a separate directory in
+	// the configs dir: "objects/<pkgName>/<bundleName>/"
+	CsvJSON string   `json:"-"`
+	Objects []string `json:"-"`
+}
+
+type RelatedImage struct {
+	Name  string `json:"name"`
+	Image string `json:"image"`
+}
+
+type Meta struct {
+	Schema  string
+	Package string
```

Comment:
You'll probably want to put these comments in #601, since that's where the `pkg/declcfg` originates.

But to answer this question... `Package` is optional in the meta blob, and is helpful to align package-specific blobs with their `olm.package`. There's a blurb about this in the EP [here](https://github.com/operator-framework/enhancements/blob/master/enhancements/declarative-index-config.md#filesystem-structure).